dojo.py

```diff
@@ -18,93 +19,85 @@ def create_room(self, room_type, room_names):
-			for room in self.rooms:
-				current_rooms.append(room['room_name'])
-			if room_name in current_rooms:
+			found = False
+			for room in self.all_rooms:
+				if room_name == room.room_name:
+					found = True
+			if found:
 				msg += "Room with name " + room_name + " already exists \n"
 			else:
 				if room_type in ('office', 'living'):
-					self.rooms.append({'room_name': room_name, 'occupants': occupants, 'type': room_type, 'room_capacity': room_capacity})
-					if room_type == 'office':
-						room = Office(room_name)
-						self.offices.append({'room_name': room_name, 'occupants': occupants, 'room_capacity': room_capacity})
-						msg += "Office " + room_name + " added successfully \n"
-					elif room_type == 'living':
-						room = LivingSpace(room_name)
-						self.livingSpaces.append({'room_name': room_name, 'occupants': occupants, 'room_capacity': room_capacity})
-						msg += "Living space " + room_name + " added successfully \n"
+					room = Room(room_name)
+					self.all_rooms.append(room)
+				if room_type == 'office':
+					room = Office(room_name)
+					self.offices.append(room)
+					msg += "Office " + room_name + " added successfully \n"
+				elif room_type == 'living':
+					room = LivingSpace(room_name)
+					self.livingSpaces.append(room)
+					msg += "Living space " + room_name + " added successfully\n"
 				else:
 					msg += "Wrong room type \n"
-		return msg
+			return msg
+
 
-		
-	def add_person(self, name, category, accomodation="N"):
+	def add_person(self, first_name, last_name, category, accomodation="N"):
 		""" function to add a person """
-		name = name.lower()
+		full_name = first_name.capitalize() + " " + last_name.capitalize()
 		category = category.lower()
 		accomodation = accomodation.upper()
 
-		people = []
-		for person in self.fellows:
-			people.append(person['name'])
-		for person in self.staffs:
-			people.append(person['name'])
-		if name in people:
-			choice = input(name + " has already been used, Enter Y to proceed or C to cancel:")
-			if choice.upper() not in ['Y', 'C']:
-				return "Invalid choice"
-			elif choice.upper() == 'C':
-				return "Bye"
 		if category in ("fellow", "staff"):
-			
-			id = len(self.persons) + 1
-			self.persons.append({'name': name, 'id': id})
-			if category == "fellow":
-				person = Fellow(name, accomodation)
-				self.fellows.append({'name': name, 'id': id})
-				msg = self.allocate_office(name)
-				msgs = ''
-				if accomodation == 'Y':
-					msgs = self.allocate_living(name)
-				return "Fellow added successfully with id " + str(id) + ". " + msg + ". " + msgs
 
-			elif category == "staff":
-				person = Staff(name, accomodation)
-				if accomodation == "Y":
-					return "Sorry! staff can't be accomodated" 
-				else:
-					self.staffs.append({'name': name, 'id': id})
-					msg = self.allocate_office(name)
-					return "Staff added successfully with " + str(id) + ". " + msg
+			id = len(self.all_people) + 1
+			person = Person(first_name, last_name)
```

Comment:
Again, Person is an abstract class. You can only add a fellow or a staff. Not a person. Otherwise, "Andrew" in all_people will not be the  same as "Andrew" in fellows because one is an instance of Person the other is an instance of Fellow